compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -803,6 +805,14 @@ public void visit(BLangWhile whileNode) {
                 this.uninitializedVars.put(symbol, InitStatus.PARTIAL_INIT);
             }
         }
+
+        if (whileResult.flowTerminated ||
```

Comment:
The following doesn't result in an error.

```ballerina
function fn() {
    int i;
    boolean b = false;

    while true {
        while b {
            i = 1;
        }

        break;
    }

    int _ = i;
}
```

Let's add tests for this and other nested cases also?